services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusServiceTest.kt

```diff
@@ -446,6 +451,157 @@ class HakemusServiceTest {
         }
     }
 
+    @Nested
+    inner class OperationalCondition {
```

Comment:
Is a test for the application type missing here?

Reply:
It's tested in the integration tests.

Reply:
Yes, it is, but why there and not here? I understand that in integration test one tests the ingration between services, the database and external services but checking the type of the application does not need any of those things, right? At least for me it seems a similar test than testing the dates or states which are tested here in the unit test.

Reply:
It's in integration tests because currently, there's no easy way to create a hakemus with an arbitrary type without the builder, which needs database access.

I had all these tests in the integration tests at first, but with so many parametrized tests they were slow so I moved them to unit tests.